dvc/cache/local.py

```diff
@@ -55,11 +55,6 @@ def makedirs(self, path_info):
 
         makedirs(path_info, exist_ok=True, mode=self._dir_mode)
 
-    def link(self, from_info, to_info):
-        super().link(from_info, to_info)
-        if self.cache_types[0] not in ("symlink", "hardlink"):
-            os.chmod(to_info, self._file_mode)
```

Comment:
This was only needed to put normal permissions on reflinks (as they initially adopt the original ones), moved to `reflink()`.